apis/openstack-swift/src/main/java/org/jclouds/openstack/swift/v1/blobstore/RegionScopedSwiftBlobStore.java

```diff
@@ -579,22 +590,79 @@ public long countBlobs(String containerName, ListContainerOptions options) {
       throw new UnsupportedOperationException();
    }
 
-   // copied from BaseBlobStore
+   // TODO: Move statics up to BaseBlobStore to reuse?
+   @com.google.inject.Inject(optional = true)
+   @Named("jclouds.mpu.parallel.degree")
+   @VisibleForTesting
+   private int numThreads = 16;
+
+   private ExecutorService jcloudsBigBlobUploadTreadPool = MoreExecutors.getExitingExecutorService(
```

Comment:
We should not add random `ExecutorService` to our code.  Either we reuse the deprecated `userExecutor` and `ioExecutor` or make a caller provide this via a parameter.